src/USER-MISC/fix_wall_stochastic.cpp

```diff
@@ -0,0 +1,256 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   http://lammps.sandia.gov, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "fix_wall_stochastic.h"
+#include <cstring>
+#include <cstdlib>
+#include "atom.h"
+#include "comm.h"
+#include "update.h"
+#include "modify.h"
+#include "domain.h"
+#include "lattice.h"
+#include "input.h"
+#include "variable.h"
+#include "error.h"
+#include "force.h"
+#include "random_mars.h"
+
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+
+enum{NONE=0,DIFFUSIVE=1,MAXWELL=2,CERCIGNANILAMPIS=3};
+
+
+/* ---------------------------------------------------------------------- */
+
+FixWallStochastic::FixWallStochastic(LAMMPS *lmp, int narg, char **arg) :
+  FixWallReflect(lmp, narg, arg), random(NULL)
+{
+  int arginc,dir;
+
+  if (narg < 8) error->all(FLERR,"Illegal fix wall/stochastic command");
+
+  dynamic_group_allow = 1;
+
+  // parse args
+
+  nwall = 0;
+  int scaleflag = 1;
+  reflectionstyle = NONE;
+
+
+  if (strcmp(arg[3],"diffusive") == 0) {
+    reflectionstyle = DIFFUSIVE;
+    arginc = 6;
+  } else if (strcmp(arg[3],"maxwell") == 0) {
+    reflectionstyle = MAXWELL;
+    arginc = 7;
+  } else if (strcmp(arg[3],"cercignanilampis") == 0) {
+    reflectionstyle = CERCIGNANILAMPIS;
+    arginc = 9;
+  } else error->all(FLERR,"Illegal fix wall/stochastic command");
+
+  if (reflectionstyle != NONE) {
+    seedfix = force->numeric(FLERR,arg[4]);
+    if (seedfix > 900000000) error->all(FLERR,"Random seed must be less than 900000000");
+    random = new RanMars(lmp,seedfix);

```

Comment:
This should be using `random = new RanMars(lmp,seedfix + comm->me);`. Or else random numbers may be correlated between different MPI ranks. See fix langevin.